src/hooks/routing/useRoutingAPITrade.ts

```diff
@@ -0,0 +1,136 @@
+import { skipToken } from '@reduxjs/toolkit/query/react'
+import { Currency, CurrencyAmount, TradeType } from '@uniswap/sdk-core'
+import { useRoutingAPIArguments } from 'hooks/routing/useRoutingAPIArguments'
+import useIsValidBlock from 'hooks/useIsValidBlock'
+// import { IMetric, MetricLoggerUnit, setGlobalMetric } from '@uniswap/smart-order-router'
+// import { sendTiming } from 'components/analytics'
+import { useStablecoinAmountFromFiatValue } from 'hooks/useUSDCPrice'
+import ms from 'ms.macro'
+import { useMemo } from 'react'
+import { useGetQuoteQuery } from 'state/routing/slice'
+import { GetQuoteResult, InterfaceTrade, TradeState } from 'state/routing/types'
+import { computeRoutes, transformRoutesToTrade } from 'state/routing/utils'
+
+/**
+ * Returns the best trade by invoking the routing api or the smart order router on the client
+ * @param tradeType whether the swap is an exact in/out
+ * @param amountSpecified the exact amount to swap in/out
+ * @param otherCurrency the desired output/payment currency
+ */
+export function useRoutingAPITrade<TTradeType extends TradeType>(
+  // TODO: fxn name is misleading: uses API & SOR
+  tradeType: TTradeType,
+  routerApiBaseUrl?: string,
+  amountSpecified?: CurrencyAmount<Currency>,
+  otherCurrency?: Currency
+): {
+  state: TradeState
+  trade: InterfaceTrade<Currency, Currency, TTradeType> | undefined
+} {
+  const [currencyIn, currencyOut]: [Currency | undefined, Currency | undefined] = useMemo(
+    () =>
+      tradeType === TradeType.EXACT_INPUT
+        ? [amountSpecified?.currency, otherCurrency]
+        : [otherCurrency, amountSpecified?.currency],
+    [amountSpecified, otherCurrency, tradeType]
+  )
+
+  const useClientSideRouter = !Boolean(routerApiBaseUrl) // False if URL is '' or undefined
```

Comment:
useClientSideRouter no longer is a user setting; just check if routerApiBaseUrl is empty. Will remove param entirely once I verify with Alex about the user setting toggle when he's back from OOO